packages/data-stores/src/plans/hooks/test/use-pricing-meta-for-grid-plans.ts

```diff
@@ -286,7 +283,119 @@ describe( 'usePricingMetaForGridPlans', () => {
+			},
+		};
+
+		expect( pricingMeta ).toEqual( expectedPricingMeta );
+	} );
+
+	it( 'should return intro offer when available', () => {
+		Plans.useIntroOffers.mockImplementation( () => ( {
+			[ PLAN_BUSINESS ]: introOffer,
+		} ) );
+		Plans.useSitePlans.mockImplementation( () => ( {
+			isLoading: false,
+			data: {
+				[ PLAN_BUSINESS ]: {
+					...SITE_PLANS[ PLAN_BUSINESS ],
+					pricing: {
+						...SITE_PLANS[ PLAN_BUSINESS ].pricing,
+						introOffer,
+					},
+				},
+			},
+		} ) );
+
+		const pricingMeta = usePricingMetaForGridPlans( {
+			planSlugs: [ PLAN_BUSINESS ],
+			storageAddOns: null,
+			siteId,
+			coupon: undefined,
+			useCheckPlanAvailabilityForPurchase,
+		} );
+
+		const expectedPricingMeta = {
+			[ PLAN_BUSINESS ]: {
+				...SITE_PLANS[ PLAN_BUSINESS ].pricing,
+				billPeriod: undefined,
+				billingPeriod: SITE_PLANS[ PLAN_BUSINESS ].pricing.billPeriod,
+				expiry: null,
+				introOffer,
+			},
+		};
+
+		expect( pricingMeta ).toEqual( expectedPricingMeta );
+	} );
+
+	it( 'should return intro offer pricing and standard pricing adjusted by storage selection', () => {
+		Plans.useIntroOffers.mockImplementation( () => ( {
+			[ PLAN_BUSINESS ]: introOffer,
+		} ) );
+		Plans.useSitePlans.mockImplementation( () => ( {
+			isLoading: false,
+			data: {
+				[ PLAN_BUSINESS ]: {
+					...SITE_PLANS[ PLAN_BUSINESS ],
+					pricing: {
+						...SITE_PLANS[ PLAN_BUSINESS ].pricing,
+						introOffer,
+					},
+				},
+			},
+		} ) );
+		useSelect.mockImplementation( ( selectFunc ) => {
+			const select = ( store ) => {
+				if ( store === WpcomPlansUI.store ) {
+					return {
+						getSelectedStorageOptions: () => ( {
+							[ PLAN_BUSINESS ]: ADD_ON_50GB_STORAGE,
+						} ),
+					};
+				}
+
+				return null;
+			};
+
+			return selectFunc( select );
+		} );
+
+		const pricingMeta = usePricingMetaForGridPlans( {
+			planSlugs: [ PLAN_BUSINESS ],
+			storageAddOns: STORAGE_ADD_ONS_MOCK,
+			siteId,
+			coupon: undefined,
+			useCheckPlanAvailabilityForPurchase,
+		} );
+
+		const expectedPricingMeta = {
+			[ PLAN_BUSINESS ]: {
+				originalPrice: {
+					full:
+						SITE_PLANS[ PLAN_BUSINESS ].pricing.originalPrice.full +
+						STORAGE_ADD_ONS_MOCK[ 0 ].prices.yearlyPrice,
+					monthly:
+						SITE_PLANS[ PLAN_BUSINESS ].pricing.originalPrice.monthly +
+						STORAGE_ADD_ONS_MOCK[ 0 ].prices.monthlyPrice,
+				},
+				discountedPrice: {
+					full:
+						SITE_PLANS[ PLAN_BUSINESS ].pricing.discountedPrice.full +
+						STORAGE_ADD_ONS_MOCK[ 0 ].prices.yearlyPrice,
+					monthly:
+						SITE_PLANS[ PLAN_BUSINESS ].pricing.discountedPrice.monthly +
+						STORAGE_ADD_ONS_MOCK[ 0 ].prices.monthlyPrice,
+				},
+				billingPeriod: SITE_PLANS[ PLAN_BUSINESS ].pricing.billPeriod,
+				currencyCode: SITE_PLANS[ PLAN_BUSINESS ].pricing.currencyCode,
+				expiry: null,
+				introOffer: {
+					...introOffer,
+					rawPrice: {
+						monthly: 12.5 + STORAGE_ADD_ONS_MOCK[ 0 ].prices.monthlyPrice,
```

Comment:
`Non-blocking / Nitpick :`

I believe that we can derive the rawPrice intro offer prices from the introOffer object https://github.com/Automattic/wp-calypso/pull/95428/files#diff-a46b47c6eea7443bc49db0374dd5b54d5a14a95c388dd9cca9481ce161ba6fcfR76-R77